csrc/codegen.cpp

```diff
@@ -3583,8 +3585,12 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     bool bidz = sync->syncDims().get(ParallelType::BIDz);
 
     ArgumentBuilder sync_call_template_parms;
-    sync_call_template_parms.arg(bidx).arg(bidy).arg(bidz).arg(true).arg(
-        isAligned());
+    sync_call_template_parms.arg(bidx)
+        .arg(bidy)
+        .arg(bidz)
+        .arg(/*PERSISTENT=*/true)
+        .arg(/*Aligned=*/
+             has_warp_specialized_ ? false : isAligned());
```

Comment:
Why is this?

Reply:
When warp specialization is used, data loading threads terminate (return) after completing all loads. These threads do not participate in subsequent computations or block/grid synchronizations. As a result, `aligned` should always be set to false. Ideally, the function `isAligned()` should detect this scenario, but currently, it does not.